packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -0,0 +1,227 @@
+
+type HoverCardContextValue = {
+  open: boolean;
+  onOpenChange(open: boolean): void;
+  onOpen(): void;
+  onClose(): void;
+};
+
+const [HoverCardProvider, useHoverCardContext] = createContext<HoverCardContextValue>(
+  HOVERCARD_NAME
+);
+
+type HoverCardOwnProps = {
+  open?: boolean;
+  defaultOpen?: boolean;
+  onOpenChange?: (open: boolean) => void;
+  openDelay?: number;
+  closeDelay?: number;
+};
+
+const HoverCard: React.FC<HoverCardOwnProps> = (props) => {
+  const {
+    children,
+    open: openProp,
+    defaultOpen,
+    onOpenChange,
+    openDelay = 700,
+    closeDelay = 300,
+  } = props;
+  const openInvokedRef = React.useRef(false);
+  const openTimerRef = React.useRef(0);
+  const closeTimerRef = React.useRef(0);
+
+  const [open = false, setOpen] = useControllableState({
+    prop: openProp,
+    defaultProp: defaultOpen,
+    onChange: onOpenChange,
+  });
+
+  const handleOpen = React.useCallback(() => {
+    clearTimeout(closeTimerRef.current);
+
+    openTimerRef.current = window.setTimeout(() => {
+      setOpen(true);
+      openInvokedRef.current = true;
+    }, openDelay);
+  }, [openDelay, setOpen]);
+
+  const handleClose = React.useCallback(() => {
+    clearTimeout(openTimerRef.current);
+
+    closeTimerRef.current = window.setTimeout(() => {
+      if (openInvokedRef.current) {
+        setOpen(false);
+        openInvokedRef.current = false;
+      }
+    }, closeDelay);
+  }, [closeDelay, setOpen]);
+
+  // cleanup any queued state updates on unmount
+  React.useEffect(() => {
+    return () => {
+      clearTimeout(openTimerRef.current);
+      clearTimeout(closeTimerRef.current);
+    };
+  }, []);
+
+  return (
+    <HoverCardProvider open={open} onOpenChange={setOpen} onOpen={handleOpen} onClose={handleClose}>
+      <PopperPrimitive.Root>{children}</PopperPrimitive.Root>
+    </HoverCardProvider>
+  );
+};
+
+HoverCard.displayName = HOVERCARD_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * HoverCardTrigger
+ * -----------------------------------------------------------------------------------------------*/
+
+const TRIGGER_NAME = 'HoverCardTrigger';
+const TRIGGER_DEFAULT_TAG = 'a';
+
+type HoverCardTriggerOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type HoverCardTriggerPrimitive = Polymorphic.ForwardRefComponent<
+  typeof TRIGGER_DEFAULT_TAG,
+  HoverCardTriggerOwnProps
+>;
+
+const HoverCardTrigger = React.forwardRef((props, forwardedRef) => {
+  const { as = TRIGGER_DEFAULT_TAG, ...triggerProps } = props;
+  const context = useHoverCardContext(TRIGGER_NAME);
+
+  return (
+    <PopperPrimitive.Anchor
+      data-state={context.open ? 'open' : 'closed'}
+      {...triggerProps}
+      as={as}
+      ref={forwardedRef}
+      onPointerEnter={composeEventHandlers(props.onPointerEnter, excludeTouch(context.onOpen))}
+      onPointerLeave={composeEventHandlers(props.onPointerLeave, excludeTouch(context.onClose))}
```

Comment:
Excluding internal handlers from `touch` to avoid jarring interactions on mobile devices.